src/Data/Fin/Properties.agda

```diff
@@ -53,6 +53,9 @@ open import Relation.Unary.Properties using (U?)
 Fin0↔⊥ : Fin 0 ↔ ⊥
 Fin0↔⊥ = mk↔′ ¬Fin0 (λ ()) (λ ()) (λ ())
 
+Fin1↔⊤ : Fin 1 ↔ ⊤
+Fin1↔⊤ = mk↔′ (λ { 0F → tt }) (λ { tt → 0F }) (λ { tt → refl }) λ { 0F → refl }
```

Comment:
Do we even need these to be pattern-matching functions as opposed to
```suggestion
Fin1↔⊤ = mk↔′ (const tt) (const 0F) (λ _ → refl) (λ _ → refl)
```
?

Reply:
We certainly don't need to use pattern-matching, though I personally like the explicit reminder that there is only one domain value. Are there operational differences? If we do use `const`, is there any reason not to use it in all four arguments?

If there's a standard, consistent style used in agda-stdlib for such definitions, I'm happy to adopt it.

Reply:
Since I'm a relative newcomer to Agda, I may be missing some historical, operational, or even semantic context.

Reply:
Thanks for the explanation. What a shame that these general lambdas have ergonomic drawbacks! Were they hastily implemented? Can the implementation be improved? I'm trying and failing to relate to your remark that the generated names "show up in a type". Can you give an example (copy & paste from Agda preferably)? Maybe I saw that behavior and dismissed it out of general bewilderment in using an unfamiliar-to-me language.

Reply:
We have this problem here for instance: http://agda.github.io/agda-stdlib/Codata.Stream.Properties.html#1941

If you try to replace the `_` in `as ⁺++ _ ` with the expected pattern-matching
lambda you'll get an error telling you that it's not equal to another pattern-matching
that looks exactly the same.

Reply:
> Thanks for the explanation. What a shame that these general lambdas have ergonomic drawbacks! Were they hastily implemented? Can the implementation be improved? I'm trying and failing to relate to your remark that the generated names "show up in a type". Can you give an example (copy & paste from Agda preferably)? Maybe I saw that behavior and dismissed it out of general bewilderment in using an unfamiliar-to-me language.

I'd say it isn't any more hastily implemented than most parts of Agda. The problem is a fundamental tradeoff one has to make when adding a new feature between either extending the core language (and thus increasing the "attack surface" where unsoundness could be introduced) or desugaring the new feature to existing core syntax (thus running into encoding issues like the one @gallais described here). Ideally we'd have a super-small core language with perfect abstractions on top that never leak any of their implementation details, but we're not there yet (and it is not at all obvious how to get there).

Reply:
Makes sense to me. Thanks!

Reply:
When I try the suggested replacement code (or replace `λ _ → refl` with `(const refl)`), I get the following error message:

    .../src/Data/Fin/Properties.agda:57,57-61
    0F != x of type Fin (suc _n_24)
    when checking that the expression refl has type
    const 0F (const tt x) ≡ x

Repairing this error seems to require restoring all but the third pattern lambda.